source/sync_serial.cpp

```diff
@@ -31,17 +39,33 @@ static PVoidCallback irq_callback[NUM_SYNC_SERIAL];
 static void *irq_callback_data[NUM_SYNC_SERIAL];
 static uint8_t used_serials;
 
+// To be able to select a specific device
+void *allocate_sync_serial(void* device)
+{
+    int i = 0;
+    while (i < NUM_SYNC_SERIAL)
+    {
+        if (!(used_serials & (1 << i)) && (serials[i].device == device))
+        {
+            used_serials |= 1 << i;
+            return serials[i].device;
+        }
+        i++;
+    }
+    return NULL;
+}
+
 void *allocate_sync_serial(SyncSerialMode mode)
 {
-    int i = NUM_SYNC_SERIAL - 1;
-    while (i >= 0)
+    int i = 0;
+    while (i < NUM_SYNC_SERIAL)
```

Comment:
can we have it go backwards again? this way the first SPI will get SPIM3 (which is faster)

Reply:
I understand your intentions. I'll fix this part.

Reply:
Done in commit c030199.